tests/SimplifyTest.elm

```diff
@@ -6959,6 +6961,1488 @@ a = False
+        \() ->
+            """module A exposing (..)
+a = List.foldl (\\x y -> x |> (||) y) False
+"""
+                |> Review.Test.run (rule defaults)
+                |> Review.Test.expectErrors
+                    [ Review.Test.error
+                        { message = "Use List.any identity instead"
+                        , details =
+                            [ "Using List.foldl (||) False is the same as using List.any identity." ]
+                        , under = "List.foldl"
+                        }
+                        |> Review.Test.whenFixed """module A exposing (..)
+a = List.any identity
+"""
+                    ]
+    , test "should replace List.foldl (\\x y -> y |> (||) x) False by List.any identity" <|
+        \() ->
+            """module A exposing (..)
+a = List.foldl (\\x y -> y |> (||) x) False
+"""
+                |> Review.Test.run (rule defaults)
+                |> Review.Test.expectErrors
+                    [ Review.Test.error
+                        { message = "Use List.any identity instead"
+                        , details =
+                            [ "Using List.foldl (||) False is the same as using List.any identity." ]
+                        , under = "List.foldl"
+                        }
+                        |> Review.Test.whenFixed """module A exposing (..)
+a = List.any identity
+"""
+                    ]
+    ]
+
+
+listFoldlAllTests : List Test
+listFoldlAllTests =
+    [ test "should replace List.foldl (&&) False by always False" <|
+        \() ->
+            """module A exposing (..)
+a = List.foldl (&&) False
+"""
+                |> Review.Test.run (rule defaults)
+                |> Review.Test.expectErrors
+                    [ Review.Test.error
+                        { message = "The call to List.foldl will result in False"
+                        , details = [ "You can replace this call by False." ]
+                        , under = "List.foldl"
+                        }
+                        |> Review.Test.whenFixed """module A exposing (..)
+a = always False
+"""
+                    ]
+    , test "should replace List.foldl (&&) False list by False" <|
+        \() ->
+            """module A exposing (..)
+a = List.foldl (&&) False list
+"""
+                |> Review.Test.run (rule defaults)
+                |> Review.Test.expectErrors
+                    [ Review.Test.error
+                        { message = "The call to List.foldl will result in False"
+                        , details = [ "You can replace this call by False." ]
+                        , under = "List.foldl"
+                        }
+                        |> Review.Test.whenFixed """module A exposing (..)
+a = False
+"""
+                    ]
+    , test "should replace List.foldl (&&) True by List.all identity" <|
+        \() ->
+            """module A exposing (..)
+a = List.foldl (&&) True
```

Comment:
I think we could use one test like this but where the list argument is specified, so we can be sure that one doesn't removed by accident. (Same for similar test suites)